report/latest_map_generated.go

```diff
@@ -42,27 +42,53 @@ func (m StringLatestMap) Size() int {
 	return len(m)
 }
 
-// Merge produces a fresh StringLatestMap containing the keys from both inputs.
+// Merge produces a StringLatestMap containing the keys from both inputs.
 // When both inputs contain the same key, the newer value is used.
+// Tries to return one of its inputs, if that already holds the correct result.
 func (m StringLatestMap) Merge(n StringLatestMap) StringLatestMap {
 	switch {
-	case m == nil:
+	case len(m) == 0:
 		return n
-	case n == nil:
+	case len(n) == 0:
 		return m
 	}
-	l := len(m)
-	if len(n) > l {
-		l = len(n)
+	lenm := len(m)
+	if len(n) > lenm {
+		m, n, lenm = n, m, len(n) //swap so m is always at least as long as n
+	} else if len(n) == lenm && m[0].Timestamp.Before(n[0].Timestamp) {
+		m, n = n, m // swap equal-length arrays so first element of m is newer
```

Comment:
The mix of `lenm` and `len(n)` is quite jarring. More of that below, where we additionally get `len(m)`. I suggest getting rid of `lenm`, hence always calling `len(..)`.

Reply:
also, it would be helpful to expand that last comment to explain that a) this is an optimisation, and b) why it's useful.

Reply:
Removed `lenm` and extended comment.